### PR TITLE
Migrate to TypeScript

### PR DESCRIPTION
**Pull request by saihaj**
- [x] Enable strict mode
- [x] compare `.d.ts` and remove them
- [ ] Use `tsc` to generate `.d.ts`
- [x] use `npm@7` to install stuff

### CODE REVIEW

File: src/node/plural.ts [hunk 1]
```diff
@@ -15,7 +17,7 @@ type PluralIdentifyingRootFieldConfig = {
     input: any,
     context: any,
     info: GraphQLResolveInfo,
-  ) => ?any,
```
Comment (IvanGoncharov) [inline src/node/plural.ts:18]: @saihaj Not necessary anymore, can you please rebase?

File: src/connection/arrayConnection.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ const PREFIX = 'arrayconnection:';
  * Creates the cursor string from an offset.
  */
 export function offsetToCursor(offset: number): ConnectionCursor {
-  return base64(PREFIX + offset);
```
Comment (IvanGoncharov) [inline src/connection/arrayConnection.ts:124]: not nessary anymore, please rebase

File: src/__tests__/starWarsData.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ export function createShip(shipName: string, factionId: string): Ship {
   allShips.push(newShip);
 
   const faction = allFactions.find((obj) => obj.id === factionId);
-  faction?.ships.push(newShip.id);
+  faction.ships.push(newShip.id);
```
Comment (IvanGoncharov) [inline src/__tests__/starWarsData.ts:58]: this change is not needed you need to enable `strictNullChecks` in `tsconfig.json` for this to pass without changes

File: .eslintrc.yml [hunk 4]
```diff
@@ -450,16 +450,16 @@ overrides:
...
       '@typescript-eslint/consistent-type-assertions':
         [error, { assertionStyle: as, objectLiteralTypeAssertions: never }]
-      '@typescript-eslint/consistent-type-definitions': off # TODO consider
-      '@typescript-eslint/consistent-type-imports': off # TODO enable after TS conversion
+      '@typescript-eslint/consistent-type-definitions': error
+      '@typescript-eslint/consistent-type-imports': error
       '@typescript-eslint/explicit-function-return-type': off # TODO consider
       '@typescript-eslint/explicit-member-accessibility': off # TODO consider
-      '@typescript-eslint/explicit-module-boundary-types': off # TODO consider
-      '@typescript-eslint/member-ordering': off # TODO consider
```
Comment (IvanGoncharov) [inline .eslintrc.yml:517]: please don't change `TODO consider` stuff

File: src/node/plural.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ import type {
   GraphQLResolveInfo,
 } from 'graphql';
 
-type PluralIdentifyingRootFieldConfig = {
```
Comment (IvanGoncharov) [inline src/node/plural.ts:10]: can you please extract `type => interface` in separate commit?

File: src/__tests__/starWarsData.ts [hunk 6]
```diff
@@ -6,7 +6,7 @@
  * JSON objects in a more complex demo.
  */
 
-interface Ship {
+export interface Ship {
```
Comment (IvanGoncharov) [inline src/__tests__/starWarsData.ts:9]: why do you need this export?

File: src/__tests__/starWarsSchema.ts [hunk 7]
```diff
@@ -128,7 +128,7 @@ const { nodeInterface, nodeField } = nodeDefinitions(
  *     name: String
  *   }
  */
-const shipType = new GraphQLObjectType({
+const shipType: GraphQLObjectType = new GraphQLObjectType({
```
Comment (IvanGoncharov) [inline src/__tests__/starWarsSchema.ts:131]: is it required?
I thought TS can figure it out on its own.

File: src/__tests__/starWarsData.ts (hunk 6 above)
Comment (saihaj) [reply src/__tests__/starWarsData.ts:9]: I am as trying something and forgot to revert this

**Pull request merged**